hilti/include/rt/util.h

```diff
@@ -520,10 +520,16 @@ struct is_tuple<std::tuple<T...>> : std::true_type {};
 template<typename>
 struct is_optional : std::false_type {};
 
-/** Checks if a type is a optional. */
+/** Checks if a type is an optional. */
 template<typename... T>
 struct is_optional<std::optional<T...>> : std::true_type {};
 
+template<typename... T>
+struct is_optional<const std::optional<T...>> : std::true_type {};
+
+template<typename... T>
+struct is_optional<const std::optional<T...>&> : std::true_type {};
+
```

Comment:
I went with manual specializations since we do not use `volatile` in Spicy and the number of permutations is still somewhat manageable. Another approach would be to make `is_optional` a templated constexpr function which can then return different types based on the result of applying `std::remove_cv` and `std::remove_reference`.